src/NServiceBus.Core/Performance/MessageDurability/DetermineMessageDurabilityBehavior.cs

```diff
@@ -22,7 +21,7 @@ public override Task Invoke(OutgoingLogicalMessageContext context, Func<Task> ne
             {
                 context.AddDeliveryConstraint(new NonDurableDelivery());
 
-                context.SetHeader(Headers.NonDurableMessage,true.ToString());
+                context.Headers[Headers.NonDurableMessage] = true.ToString();
```

Comment:
interesting, since the xml spec states that bool only allows {true, false, 1, 0}. Raise an issue?